### ISSUE TITLE
Exclusion pattern generation bug

### ISSUE THREAD

**Issue by yaSebastian** [rhalaly/scope-to-this-vscode#11]
I have a workspace with 4 folders added to it:

- mod
- mod-adjustments
- mod-adjustments_0.1.2
- mod-cheats

Both *mod-adjustments* and *mod-cheats* have a subfolder *src*.

Scoping to *src* in the *mod-adjustments* folder works fine and creates the following exclusion rules:
```json
"files.exclude": {
  ...
  "[!s]*/**": true,
  "s[!r]*/**": true,
  "sr[!c]*/**": true
}
```

Scoping to the *src* in the *mod-cheats* folder doesn't work and creates weird exclusion rules:
```json
"files.exclude": {
  ...
  "[!-]*/**": true,
  "-[!c]*/**": true,
  "-c[!h]*/**": true,
  "-ch[!e]*/**": true,
  "-che[!a]*/**": true,
  "-chea[!t]*/**": true,
  "-cheat[!s]*/**": true,
  "-cheats/[!s]*/**": true,
  "-cheats/s[!r]*/**": true,
  "-cheats/sr[!c]*/**": true
}
```

I've tried renaming the *src* folder in *mod-adjustments* as well as in *mod-cheats* separately, but that didn't change anything. I also removed the *mod* folder temporarily, assuming that might cause the rules to just start with *-cheats*.
Nothing changed the behaviour.
Just in case anyone questions it: *mod* and *mod-adjustments_0.1.2* do not have a *src* folder.

Any ideas what's going on here?

**Comment by rhalaly** [comment]
Hi @yaSebastian, thank you for your report. Unfortunately, I can't reproduce the issue on my end. I tried to create the same folder hierarchy, and scoped `mod-cheats/src` and got a valid pattern.

<img width="693" alt="image" src="https://user-images.githubusercontent.com/8597700/210139604-253f0ad5-e66f-4806-96f4-15b2afb2fa77.png">

Please make sure you don't have any special characters in the folder name that may affect the path string.

In addition, since your folders shared the same prefix, please note that one of the limitations we have (due to VSCode) is that folders and files that have the same prefix as the selected folder name may still appear - that's the reason that the folder `mod` is still visible even when you scope `mod-cheats/src`.

**Comment by yaSebastian** [comment]
Hi @rhalaly,

**TL;DR**
**The problems seems to come from working with an unsaved workspace, disappearing when the workspace is saved to a file.**

Generally, I use workspaces in VS Code just as a temporary thing, mostly for the convenience of having related folders open side by side in the same window. Losing them would be just a minor inconvenience and I'd restore them manually, but that hasn't happened a single time over all the years and VS Code updates, so thinking about that as a source of the problem not even remotely entered my mind.
Thanks to your screenshot it did. Once I stored the workspace to disk, the issue disappeared.

I double checked for any special character names and couldn't find any. I also did a few quick random tests to see if I could find any other pattern, but couldn't find any besides saving/not saving the workspace.

As for the limitations, they are fine, one side effect is even highly desirable in my current use case. Having not just one but both projects/folders show their /src.

Thanks for your reply even over the holidays. And thanks for your extension, it's awesome, it helps so much to focus in bigger projects, reducing the cognitive load by hiding all the clutter!
